desktop/libs/indexer/src/indexer/templates/importer.mako

```diff
@@ -229,6 +229,14 @@ ${ commonheader(_("Importer"), "indexer", user, request, "60px") | n,unicode }
             <!-- /ko -->
           </div>
 
+          <div data-bind="visible: createWizard.source.inputFormat() == 'file from local'">
+              <form method="POST" action="${url('indexer:indexer_api3_upload')}" enctype="multipart/form-data">
```

Comment:
Then we could trigger the guess_format(), but this one will need to know the name. Upload could return a uuid, or probably better guess_format() could handle the upload instead of indexer_api3_upload()?

Reply:
when I am uploading a file it is redirecting to `http://127.0.0.1:8000/indexer/api/importer/upload`

and also tried with the adding code of upload to guess_formate but in this also case, the same is happening.

What I understood is that we need the path of file and then it will be equal to createWizard.source.path then after it automatically triggers the guess_formate API?
https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/templates/importer.mako#L1777-L1782

 @romainr Not getting the idea of how to proceed next.

Reply:
You need to post the data via JS, regular HTML form upload redirects.

https://www.google.com/search?q=html+upload+file+without+redirect&oq=upload+file+not+red&aqs=chrome.2.69i57j0i22i30l4.8411j0j1&sourceid=chrome&ie=UTF-8